core/storage.go

```diff
@@ -55,20 +60,20 @@ type ObjectIter interface {
 	Close()
 }
 
-// TxObjectStorage is an in-progress storage transaction.
-// A transaction must end with a call to Commit or Rollback.
-type TxObjectStorage interface {
-	Set(Object) (Hash, error)
-	Get(ObjectType, Hash) (Object, error)
+// TxObjectStorer is an in-progress storage transaction. A transaction must end
+// with a call to Commit or Rollback.
+type TxObjectStorer interface {
```

Comment:
We have an ObjectStorerTx  interface and a TxObjectStorer interface.

We must step back, think and organize the code.